ouroboros-consensus/src/Ouroboros/Consensus/Protocol/ModChainSel.hs

```diff
@@ -67,3 +70,26 @@ instance (Typeable p, Typeable s, ChainSelection p s) => OuroborosTag (ModChainS
 
 instance OuroborosTag p => NoUnexpectedThunks (NodeConfig (ModChainSel p s))
   -- use generic instance
+
+{-------------------------------------------------------------------------------
+  Individual modifiers
+-------------------------------------------------------------------------------}
+
+-- | Override 'preferCandidate' such that the ChainDB will select a new EBB
+-- /promptly/.
+--
+-- Uses the default 'compareCandidates' from the definition of 'OuroborosTag'.
+data SelectEBBsPromptly
```

Comment:
I'm not entirely convinced we should add this as a modifier. My thoughts:
* Edsko recently removed most modifiers as they were making things more complex.
* (Similar:) while modifiers can be elegant, maybe it would be simpler if we just inlined it in `PBFT`?
* ~Only RealPBFT needs to deal with EBBs, so the combinator is not really reusable anyway.~ We could use it for MockPBFT. Moreover, EBBs will always be in the chain, so all future protocols will have to deal with it, at least in case of corruptions, such that the chain is truncated into an era that still has EBBs. I suppose this means that we should use `compareHeadBlockNoThenLength` (see comment below) as the default for all protocols.

Reply:
We chatted on Slack. Thomas indeed wants to adopt this improved `preferCandidates` (with a matching `compareCandidates`) in `PBFT` itself. Seems like the previous half-support for EBBs (ie "we'll only select an EBB once its successor arrives") was itself a compromise, and the new `intersection`-based `preferCandidates` is itself desirable.

Thus: no combinator: just bake it in `PBFT` and also update `compareCandidates`.

Reply:
OK, I updated the default `compareCandidates` to compare by block number and then by length-after-intersection. All protocols use this default.

I also updated the default `preferCandidates` to delegate to `compareCandidates`. All protocols except Praos use this default, but I also updated its custom `preferCandidates` to use `compareCandidates`.

I also needed to update `BlockFetch.Decision.ChainSuffix` to carry around the original candidate so that we could compare based on that field. Before this, it only carried the blocks after the intersection with the current chain, which might not themselves intersect. This new field is guaranteed to intersect among the candidates, since this is downstream of the "filter plausible" etc. After this, it seems like the new PRECONDITION on `compareCandidates` (that they intersect, at the very least when the block number is equal) is satisfied.

Reply:
Resolution: we backed this out and used the simpler scheme that Edsko and Thomas came up with.